src/state_machine/coordinator/fire.rs

```diff
@@ -1680,12 +1690,47 @@ pub mod test {
             msg: Message::SignatureShareResponse(sig_share_response.clone()),
             sig: Default::default(),
         };
+
+        let response_info = coordinator
+            .message_nonces
+            .entry(coordinator.message.clone())
+            .or_default();
+        response_info.sign_wait_signer_ids.insert(0);
+
         coordinator
             .gather_sig_shares(&packet, signature_type)
             .unwrap();
         assert_eq!(1, coordinator.signature_shares.len());
 
-        // check that a duplicate private share is ignored
+        // check that a signature share is ignored due to not being on the wait list
+        let dup_signature_share = SignatureShare {
+            id: 1,
+            z_i: Scalar::random(&mut rng),
+            key_ids: vec![1],
+        };
+        let dup_sig_share_response = SignatureShareResponse {
+            dkg_id: 0,
+            sign_id: 0,
+            sign_iter_id: 0,
+            signer_id: 0,
+            signature_shares: vec![dup_signature_share.clone()],
+        };
+        let packet = Packet {
+            msg: Message::SignatureShareResponse(dup_sig_share_response.clone()),
+            sig: Default::default(),
+        };
+        coordinator
+            .gather_sig_shares(&packet, signature_type)
+            .unwrap();
+        assert_eq!(1, coordinator.signature_shares.len());
+
+        // check that a duplicate signature share is ignored even if on the wait list
+        let response_info = coordinator
+            .message_nonces
+            .entry(coordinator.message.clone())
+            .or_default();
+        response_info.sign_wait_signer_ids.insert(0);
```

Comment:
qq: why not reusing the same package below (cannot comment that line since it's not part of the diff)? It seems to be exactly the same

Reply:
I got rid of one of the `dup_sig_share_response` vars in response to the previous comment, but I still have one at the end.  It's not exactly the same, it has a different `z_i` value.